### PR TITLE
Cure unphysical behaviour of YD-switch

### PR DESCRIPTION
**Pull request by AHaumer**
Electrical.Machines.Utilities.SwitchYD produces during switching from Y-connection to D-connection unphysical voltage peaks, especially visible when inductors are placed between the voltage source and the machine. Cured by offering an additional SwitchYD with arc.

### CODE REVIEW

File: Modelica/Electrical/Machines/Examples/InductionMachines/IMC_YDarc.mo [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    annotation (Line(points={{-20,90},{0,90}}, color={0,0,255}));
+  connect(resistor.plug_n, inductor.plug_p)
+    annotation (Line(points={{0,70},{0,60}}, color={0,0,255}));
+  connect(inductor.plug_n, idealCloser.plug_p)
+    annotation (Line(points={{0,40},{0,30}}, color={0,0,255}));
+  annotation (experiment(
+      StopTime=2.5,
+      Interval=0.0001,
+      Tolerance=1e-06,
+      __Dymola_Algorithm="Dassl"),                                      Documentation(
```
Comment (beutlich) [inline Modelica/Electrical/Machines/Examples/InductionMachines/IMC_YDarc.mo]: This should be removed.

Comment (AHaumer) [reply Modelica/Electrical/Machines/Examples/InductionMachines/IMC_YDarc.mo]: What do you mean by "This"?

Comment (beutlich) [reply Modelica/Electrical/Machines/Examples/InductionMachines/IMC_YDarc.mo]: Please delete `__Dymola_Algorithm="Dassl"`.

File: Modelica/Electrical/Machines/Examples/InductionMachines/IMC_YDarc.mo [hunk 2]
```diff
@@ -125,11 +125,7 @@ equation
     annotation (Line(points={{0,70},{0,60}}, color={0,0,255}));
   connect(inductor.plug_n, idealCloser.plug_p)
     annotation (Line(points={{0,40},{0,30}}, color={0,0,255}));
-  annotation (experiment(
-      StopTime=2.5,
-      Interval=0.0001,
-      Tolerance=1e-06,
-      __Dymola_Algorithm="Dassl"),                                      Documentation(
+  annotation (experiment(StopTime=2.5, Interval=0.0001, Tolerance=1e-06,), Documentation(
```
Comment (beutlich) [inline Modelica/Electrical/Machines/Examples/InductionMachines/IMC_YDarc.mo]:
```suggestion
  annotation (experiment(StopTime=2.5, Interval=0.0001, Tolerance=1e-06), Documentation(
```

Comment (AHaumer) [reply Modelica/Electrical/Machines/Examples/InductionMachines/IMC_YDarc.mo]: resolved by ea7c5d611..8d1bf4e3f

File: Modelica/Electrical/Machines/Utilities/LogicalDelay.mo [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  extends Blocks.Icons.BooleanBlock;
+  parameter SIunits.Time delayTime(final min=0)=0 "Time delay";
+  Blocks.Interfaces.BooleanInput u
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+  Blocks.Interfaces.BooleanOutput y1
+    annotation (Placement(transformation(extent={{100,50},{120,70}})));
+  Blocks.Interfaces.BooleanOutput y2
+  annotation (Placement(transformation(extent={{100,-70},{120,-50}})));
+protected
+  Boolean not_u=not u;
```
Comment (dietmarw) [inline Modelica/Electrical/Machines/Utilities/LogicalDelay.mo]: It looks unnecessary to introduce `not_u` when you could just write `not u` in the two places this is used.

Comment (AHaumer) [reply Modelica/Electrical/Machines/Utilities/LogicalDelay.mo]: resolved by 8d1bf4e3f..b3435eb49

**Comment by AHaumer** [comment]
@dietmarw done by b3435eb49..11247c07b

**Comment by dietmarw** [comment]
I don't think this qualifies as a bug as the models don't have any error in them. Even the YD-Starter example does not create any problem if used as is. Only if one modifies the example model the voltage spikes will occur. But that is then on the users side and how the user implements the existing models. The new models are useful and is a nice extension but do not qualify as back-porting bug fix.

**Pull request merged**